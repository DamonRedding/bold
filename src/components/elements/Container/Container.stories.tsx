import { storiesOf } from '@storybook/react'
import React from 'react'

import { Container } from './Container'

storiesOf('Grid', module).add('Container', () => <Container style={{ background: '#f2f2f2' }}>Content</Container>)
