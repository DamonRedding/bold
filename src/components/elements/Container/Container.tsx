import { Interpolation } from 'emotion'
import React, { CSSProperties } from 'react'

import { Theme, useStyles } from '../../../styles'
import { Omit } from '../../../util'

export interface ContainerProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'style'> {
  style?: Interpolation
}

export function Container(props: ContainerProps) {
  const { style, ...rest } = props
  const { classes, css } = useStyles(createStyles)

  return <div className={css(classes.container, style)} {...rest} />
}

Container.defaultProps = {} as Partial<ContainerProps>

export const createStyles = (theme: Theme) => ({
  container: {
    width: '960px',
    margin: '0 auto',

    [theme.breakpoints.down('md')]: {
      width: '768px',
    },
  } as CSSProperties,
})
