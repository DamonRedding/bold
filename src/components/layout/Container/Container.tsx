import { Interpolation } from 'emotion'
import React from 'react'

import { withStyles, WithStylesProps } from '../../../styles'

export interface ContainerProps extends WithStylesProps {
    style?: Interpolation
}

@withStyles
export class Container extends React.PureComponent<ContainerProps> {

    static defaultProps: Partial<Container> = {}

    render() {
        const { theme, style } = this.props
        const styles = {
            container: {
                width: '960px',
                margin: '0 auto',

                [theme.breakpoints.down('small')]: {
                    width: '768px',
                },
            },
        }

        return (
            <div className={this.props.css(styles.container, style)}>{this.props.children}</div>
        )
    }
}
