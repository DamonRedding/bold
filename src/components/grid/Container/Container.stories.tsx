import { storiesOf } from '@storybook/react'
import * as React from 'react'

import { withPropTypes, withTheme } from '../../../stories-addons'

import { Container } from './Container'

storiesOf('Grid', module)
    .addDecorator(withPropTypes())
    .addDecorator(withTheme())
    .add('Container', () => (
        <Container style={{ background: '#f2f2f2' }}>
            Content
        </Container>
    ))
