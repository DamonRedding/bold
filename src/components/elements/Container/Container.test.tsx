import React from 'react'
import { render } from 'react-testing-library'

import { Container } from './Container'

it('should render correctly', () => {
  const { container } = render(<Container>Content</Container>)
  expect(container).toMatchSnapshot()
})

it('should accept the style prop', () => {
  const { container } = render(<Container style={{ padding: '5rem' }}>Content</Container>)
  expect(container).toMatchSnapshot()
})

it('should accept HTML div element props', () => {
  const { container } = render(<Container id='test'>Content</Container>)
  expect(container.querySelector('div').getAttribute('id')).toEqual('test')
})
