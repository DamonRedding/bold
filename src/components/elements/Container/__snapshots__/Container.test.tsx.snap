// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should accept the style prop 1`] = `
.emotion-0 {
  width: 960px;
  margin: 0 auto;
  padding: 5rem;
}

@media (max-width:1280px) {
  .emotion-0 {
    width: 768px;
  }
}

<div
  class="emotion-0"
>
  Content
</div>
`;

exports[`should render correctly 1`] = `
.emotion-0 {
  width: 960px;
  margin: 0 auto;
}

@media (max-width:1280px) {
  .emotion-0 {
    width: 768px;
  }
}

<div
  class="emotion-0"
>
  Content
</div>
`;
